indigo_app/templates/indigo_api/_task_detail.html

```diff
@@ -191,124 +191,168 @@ <h5>
-              <button class="btn btn-outline-danger mb-3" type="submit"
-                      data-confirm="{% trans 'Are you sure you want to unassign this task?' %}"
-                      formaction="{% url 'unassign_task' place=place.place_code pk=task.pk %}">
-                {% trans 'Unassign' %}
-              </button>
+            {% if perms.indigo_api.change_task %}
+              <form method="POST" action="{% url 'assign_task' place=place.place_code pk=task.pk %}">
+                {% csrf_token %}
+                {% if task.assigned_to %}
+                  <button class="btn btn-outline-danger mb-3" type="submit"
+                          data-confirm="{% trans 'Are you sure you want to unassign this task?' %}"
+                          formaction="{% url 'unassign_task' place=place.place_code pk=task.pk %}">
+                    {% trans 'Unassign' %}
+                  </button>
+                {% endif %}
+
+                <div class="dropdown">
+                  <button class="btn btn-outline-secondary mb-3 dropdown-toggle" type="button"
+                          data-bs-toggle="dropdown"
+                          hx-trigger="click once"
+                          hx-get="{% url 'assign_task_to' place=place.place_code pk=task.pk %}"
+                          hx-target="next .dropdown-menu">
+                    {% trans 'Assign to' %}
+                  </button>
+                  <div class="dropdown-menu"><div class="dropdown-item">...</div></div>
+                </div>
+              </form>
             {% endif %}
+          </div>
+        </div>
 
-            <div class="dropdown">
-              <button class="btn btn-outline-secondary mb-3 dropdown-toggle" type="button"
-                      data-bs-toggle="dropdown"
-                      hx-trigger="click once"
-                      hx-get="{% url 'assign_task_to' place=place.place_code pk=task.pk %}"
-                      hx-target="next .dropdown-menu">
-                {% trans 'Assign to' %}
-              </button>
-              <div class="dropdown-menu"><div class="dropdown-item">...</div></div>
+        <div class="card mb-3">
+          <h6 class="card-header">{% trans 'Labels' %}</h6>
+          <div class="card-body">
+            <select
+                class="selectpicker" multiple
+                hx-trigger="change"
+                hx-post="{% url 'task_labels' place=place.place_code pk=task.pk %}"
+                hx-target="#task-labels-list"
+                name="labels"
+                data-style="btn-link ps-0 text-dark"
+                {% if not perms.indigo_api.change_task %}disabled{% endif %}
+                data-none-selected-text="{% trans 'Choose labels' %}"
+                data-selected-text-format="static"
+                data-size="7"
+            >
+              {% for label in possible_labels %}
+                <option value="{{ label.id }}" {% if label in task.labels.all %}selected{% endif %}>
```

Comment:
this re-requires the db for every option. you can either load the list once with a `with ...` or, pre-populate a TaskForm and then you can just iterate over `options` and it'll do the right thing, including knowing which are selected.

Reply:
I'd like to go with the second option but I'm actually unclear on how to do either of them 😅

Reply:
simply putting a `form` in the template uses the CommentForm by default, and I'm not sure how to change that?